carbonmark-api/src/utils/helpers/users.utils.ts

```diff
@@ -78,5 +84,21 @@ export const getProfileByHandle = async (params: {
   if (snapshot.empty) return null;
   // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- apply known type
   const profile = snapshot.docs.at(0)?.data() as UserProfile | undefined;
-  return profile || null;
+  return profile ? getUiUserProfile(profile) : null;
 };
+
+/**
+ * Converts backend profile information into frontend profile information
+ * @param timestamp
+ * @returns
+ */
+export function getUiUserProfile(profile: UserProfile): UiUserProfile {
+  return {
+    address: profile.address,
+    description: profile.description,
+    handle: profile.handle,
+    username: profile.username,
+    createdAt: (profile.createdAt / 1000).toFixed(0),
+    updatedAt: (profile.updatedAt / 1000).toFixed(0),
+  };
+}
```

Comment:
Is the main purpose of this function to format the createdAt and updatedAt timestamps? 

If so it might be better to do this explicitly as a seperate utility at the point of render. This will also avoid the need for another manual type that needs to be maintained`UiUserProfile`.

Unless there's another benefit I'm missing here?

Reply:
> Is the main purpose of this function to format the createdAt and updatedAt timestamps?
Yup
> 
> If so it might be better to do this explicitly as a seperate utility at the point of render. This will also avoid the need for another manual type that needs to be maintained`UiUserProfile`.
> 
> Unless there's another benefit I'm missing here?

The thing is that user model transformation happens also in the projects route so I would need a model transformation function anyways.
`get.ts -> fetchmarketplacelisting.ts -> users.utils.ts`

And given how the existing code is structured, it would feel awkward for that route, the handler should not have to know that this particular piece of information needs conversion.

Reply:
Yah I agree I guess my point is that the UI (App or consumer of the API) should be responsible for any formatting. 

The API should aim to be client agnostic. Do you know what I mean?

Reply:
I was not working on carbonmark when the project started so I do not have the history. But I noticed that timestamps are converted to strings on purpose (same for prices) so I wanted to stick with that original design.
But on principle I completly agree with you with returning timestamps as numbers and will modify the PR in that direction.
I tag @Atmosfearful just in case there is a real rationale behind this conversion for timestamps.